diff --git a/net/lighttpd/Makefile b/net/lighttpd/Makefile
index 7519c07..495ea38 100644
--- a/net/lighttpd/Makefile
+++ b/net/lighttpd/Makefile
@@ -8,12 +8,16 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=lighttpd
-PKG_VERSION:=1.4.42
-PKG_RELEASE:=1
-
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
-PKG_SOURCE_URL:=http://download.lighttpd.net/lighttpd/releases-1.4.x
-PKG_MD5SUM:=53c55d7e1dac7adec161cd5490491f6d
+PKG_VERSION:=1.4.40
+PKG_RELEASE:=3
+
+PKG_HASH:=c46f0ce027155924a53bf12cb306b3afb77bbd65
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)
+PKG_SOURCE_URL:=https://github.com/lighttpd/lighttpd1.4.git
+PKG_SOURCE_VERSION:=$(PKG_HASH)
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_HASH).tar.gz
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
 
 PKG_LICENSE:=BSD-3c
 PKG_LICENSE_FILES:=COPYING
